### PR TITLE
Added ChatStyle

### LINKED ISSUE #5: Make chat colours custom
Nowadays, chat use primary and accent colours as base, but it's not always true. Sometimes you want to use a different color to your chat page, or just to your chat response background.

> Would be awesome have a ChatStyle or something to define the custom style

Something like: `AppBarSettings` -> [here](https://github.com/BeeTech-global/flutter_zendesk_chat/blob/master/lib/widgets/zendesk_chat_screen.dart) that must be a model.

### PR DESCRIPTION
**Pull request by davidfranquet**
Fixes #5 
Added customisable chat style.

You can create styles and use the `copyWith()` to change some parameters for example:

image

For example in this case the secondary style will get all the main style with the main color changed

### CODE REVIEW

File: lib/widgets/zendesk_chat_widget.dart [hunk 1]
```diff
@@ -3,14 +3,66 @@ import 'package:zendesk_chat/models/chat_settings_model.dart';
 import 'package:zendesk_flutter_plugin/chat_models.dart';
 import 'package:zendesk_flutter_plugin/zendesk_flutter_plugin.dart';
 
+class ChatStyle {
```
Comment (zeucxb) [inline lib/widgets/zendesk_chat_widget.dart]: Remove from this file, please create at models

File: lib/widgets/zendesk_chat_widget.dart [hunk 2]
```diff
@@ -94,17 +150,12 @@ class _ZendeskChatWidgetState extends State<ZendeskChatWidget> {
...
-                      ),
+                      style: chatStyle.nameTextStyle,
                     ),
                   ),
                   Text(
                     chatItem?.message ?? '',
-                    style: TextStyle(
-                      color: Colors.white,
-                    ),
+                    style: chatStyle.descriptionTextStyle,
```
Comment (zeucxb) [inline lib/widgets/zendesk_chat_widget.dart]: Isn't messageTextStyle a better name?

**Review by zeucxb — CHANGES_REQUESTED** [review]

File: lib/widgets/zendesk_chat_widget.dart (hunk 1 above)
Comment (davidfranquet) [reply lib/widgets/zendesk_chat_widget.dart]: It is not a model itself, It is a style class only used on this widget so I think that it is ok to leave it here

File: lib/widgets/zendesk_chat_widget.dart (hunk 2 above)
Comment (davidfranquet) [reply lib/widgets/zendesk_chat_widget.dart]: yep

**Comment by davidfranquet** [comment]
Done

File: lib/widgets/zendesk_chat_widget.dart [hunk 3]
```diff
@@ -1,16 +1,43 @@
...
+
+  const _DefaultStyle({
+    Color mainColor,
+    Color visitorColor,
+    TextStyle nameTextStyle,
+    TextStyle messageTextStyle,
+  });
+}
+
+const ChatStyle _defaultStyle = const _DefaultStyle();
```
Comment (zeucxb) [inline lib/widgets/zendesk_chat_widget.dart:29]: With this approach if I set just `mainColor` in the `chatStyle` all the defaultStyle will be replaced, it's ok. But in this case ChatStyle properties must be required and `assert` to not null.

File: lib/models/chat_style.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    TextStyle messageTextStyle,
+  }) {
+    return ChatStyle(
+      mainColor: mainColor ?? this.mainColor,
+      visitorColor: visitorColor ?? this.visitorColor,
+      nameTextStyle: nameTextStyle ?? this.nameTextStyle,
+      messageTextStyle: messageTextStyle ?? this.messageTextStyle,
+    );
+  }
+}
```
Comment (zeucxb) [inline lib/models/chat_style.dart:29]: [LINT]: Missing new line at the end.

File: lib/widgets/zendesk_chat_widget.dart [hunk 5]
```diff
@@ -1,16 +1,43 @@
 import 'package:flutter/material.dart';
 import 'package:zendesk_chat/models/chat_settings_model.dart';
+import 'package:zendesk_chat/models/chat_style.dart';
 import 'package:zendesk_flutter_plugin/chat_models.dart';
 import 'package:zendesk_flutter_plugin/zendesk_flutter_plugin.dart';
 
+
+
+class _DefaultStyle extends ChatStyle {
+  final Color mainColor = Colors.blue;
```
Comment (zeucxb) [inline lib/widgets/zendesk_chat_widget.dart:10]: DefaultStyle must follow theme colours.

File: lib/widgets/zendesk_chat_widget.dart [hunk 6]
```diff
@@ -94,17 +125,12 @@ class _ZendeskChatWidgetState extends State<ZendeskChatWidget> {
                     margin: EdgeInsets.only(bottom: 5),
                     child: Text(
                       chatItem?.displayName,
-                      style: TextStyle(
-                        fontWeight: FontWeight.bold,
-                        color: Colors.white,
-                      ),
+                      style: chatStyle.nameTextStyle,
```
Comment (zeucxb) [inline lib/widgets/zendesk_chat_widget.dart:128]: Will it be the same for visitor and main?

File: lib/widgets/zendesk_chat_widget.dart [hunk 7]
```diff
@@ -94,17 +125,12 @@ class _ZendeskChatWidgetState extends State<ZendeskChatWidget> {
...
                     ),
                   ),
                   Text(
                     chatItem?.message ?? '',
-                    style: TextStyle(
-                      color: Colors.white,
-                    ),
+                    style: chatStyle.messageTextStyle,
```
Comment (zeucxb) [inline lib/widgets/zendesk_chat_widget.dart:133]: Same here: Will it be the same for visitor and main?

**Review by zeucxb — CHANGES_REQUESTED** [review]

**Comment by zeucxb** [comment]
@DavidFD005 Thank you for your help. And sorry for being boring 😞

File: lib/widgets/zendesk_chat_widget.dart (hunk 1 above)
Comment (zeucxb) [reply lib/widgets/zendesk_chat_widget.dart]: But it's the package core widget. And it need to be exported.

**Comment by zeucxb** [comment]
@DavidFD005 One last thing. You need to add chat_settings_model to models file to export. It would be nice to use ChatStyle in example, so you can check if it's everything exported and ok. 👌

**Comment by davidfranquet** [comment]
I will do all this stuff today

**Review by lucasdaiki — APPROVED** [review]

**Review by lucasdaiki — APPROVED** [review]

**Review by lucasdaiki — CHANGES_REQUESTED** [review]
❤️

**Review by lucasdaiki — APPROVED** [review]

**Comment by zeucxb** [comment]
@DavidFD005 => Are you still working on it?

**Pull request closed without merging**